packages/editor-ui/src/components/NodeSettings.vue

```diff
@@ -281,6 +326,34 @@ export default mixins(
 			},
 		},
 		methods: {
+			/**
+			 * Check if any of the node's credential types may be used to
+			 * make a request with the HTTP Request node v2.
+			 */
+			checkHttpRequestNodeSupport(credentialTypeNames: string[]) {
+				this.isSupportedByHttpRequestNode = credentialTypeNames.some(name => {
+					const credentialType = this.getCredentialTypeByName(name);
+
+					return (
+						credentialType.name.slice(0, -4).endsWith('OAuth') ||
+						credentialType.authenticate !== undefined
+					);
+				});
+			},
+			async prepareScopesNotice(activeCredentialType: string) {
+				if (!this.isHttpRequestNodeV2(this.node)) return;
+
+				if (!activeCredentialType || !activeCredentialType.endsWith('OAuth2Api')) {
+					this.scopes = [];
+					return;
+				}
+
+				this.scopes = await this.restApi().getScopes(activeCredentialType);
```

Comment:
this will make an api request every time the modal is opened or the parameter list refreshes. is this necessary? should we instead store some of this info in store?